packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -429,20 +435,38 @@ async function* handleSSGRoute(
     const parameters = await runInInjectionContext(parentInjector, () => getPrerenderParams());
     try {
       for (const params of parameters) {
-        const routeWithResolvedParams = currentRoutePath.replace(URL_PARAMETER_REGEXP, (match) => {
-          const parameterName = match.slice(1);
-          const value = params[parameterName];
-          if (typeof value !== 'string') {
+        const isParamsArray = Array.isArray(params);
+
+        if (isParamsArray) {
+          if (!isCatchAllRoute) {
             throw new Error(
-              `The 'getPrerenderParams' function defined for the '${stripLeadingSlash(currentRoutePath)}' route ` +
-                `returned a non-string value for parameter '${parameterName}'. ` +
-                `Please make sure the 'getPrerenderParams' function returns values for all parameters ` +
-                'specified in this route.',
+              `The 'getPrerenderParams' function for the '${stripLeadingSlash(currentRoutePath)}' ` +
+                `route returned an array '${JSON.stringify(params)}', which is not valid for catch-all routes.`,
             );
           }
+        } else if (isCatchAllRoute) {
+          throw new Error(
+            `The 'getPrerenderParams' function for the '${stripLeadingSlash(currentRoutePath)}' ` +
+              `route returned an object '${JSON.stringify(params)}', which is not valid for parameterized routes.`,
+          );
+        }
 
-          return value;
-        });
+        const routeWithResolvedParams = isParamsArray
+          ? currentRoutePath.replace('**', params.join('/'))
```

Comment:
Consider: What if I render `['foo/bar']`?

I remember we had a discussion on this previously, do we need to escape `/` characters returned by `getPrerenderParams`? Or did we decide not to do that?

Reply:
No we are not escaping any character. `foo/bar` will be rendered it's a valid URL.